diff --git a/external/arch/arm/arm-m/CMSIS/Include/cmsis_armclang.h b/external/arch/arm/arm-m/CMSIS/Include/cmsis_armclang.h
index 691141774..6420ea1b8 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/cmsis_armclang.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/cmsis_armclang.h
@@ -27,6 +27,9 @@
 #ifndef __CMSIS_ARMCLANG_H
 #define __CMSIS_ARMCLANG_H
 
+/* LK: include lk's compiler.h first, which has some of the same #defines */
+#include <lk/compiler.h>
+
 #pragma clang system_header   /* treat file as system include file */
 
 /* CMSIS compiler specific defines */
diff --git a/external/arch/arm/arm-m/CMSIS/Include/cmsis_compiler.h b/external/arch/arm/arm-m/CMSIS/Include/cmsis_compiler.h
index adbf296f1..8ac717085 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/cmsis_compiler.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/cmsis_compiler.h
@@ -47,6 +47,13 @@
   #include "cmsis_armclang.h"
 
 
+/*
+ * Any other Clang (not armclang)
+ */
+#elif defined ( __clang__ )
+  #include "cmsis_armclang.h"
+
+
 /*
  * GNU Compiler
  */
diff --git a/external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h b/external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h
index 67bda4ef3..dc3ae357f 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h
@@ -25,6 +25,9 @@
 #ifndef __CMSIS_GCC_H
 #define __CMSIS_GCC_H
 
+/* LK: include lk's compiler.h first, which has some of the same #defines */
+#include <lk/compiler.h>
+
 /* ignore some GCC warnings */
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wsign-conversion"
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h b/external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h
index 94128a1a7..272787d1a 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h
@@ -4096,7 +4096,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h b/external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h
index 932d3d188..eeb0342e9 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h
@@ -2168,7 +2168,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h b/external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h
index c119fbf24..62b6af775 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h
@@ -3077,7 +3077,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm0.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm0.h
index 6441ff341..f4d5d7e98 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm0.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm0.h
@@ -928,7 +928,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h
index 4e7179a61..7c589dc1b 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h
@@ -1063,7 +1063,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm1.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm1.h
index 76b456974..aebd6960c 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm1.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm1.h
@@ -955,7 +955,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm23.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm23.h
index 55fff9950..717bc8286 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm23.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm23.h
@@ -2243,7 +2243,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm3.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm3.h
index 74fb87e5c..fd2bcfc83 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm3.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm3.h
@@ -1841,7 +1841,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm33.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm33.h
index 18a2e6fb0..0598b93bc 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm33.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm33.h
@@ -3145,7 +3145,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h
index 3843d9542..a1f216a1b 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h
@@ -3145,7 +3145,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm4.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm4.h
index e21cd1492..52ef49224 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm4.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm4.h
@@ -2027,7 +2027,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm55.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm55.h
index faa30ce36..ee6f20892 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm55.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm55.h
@@ -4685,7 +4685,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm7.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm7.h
index 010506e9f..3a45a63bb 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm7.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm7.h
@@ -2264,7 +2264,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm85.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm85.h
index 604631118..136787605 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm85.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm85.h
@@ -4540,7 +4540,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_sc000.h b/external/arch/arm/arm-m/CMSIS/Include/core_sc000.h
index dbc755fff..fe52f94c3 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_sc000.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_sc000.h
@@ -1006,7 +1006,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_sc300.h b/external/arch/arm/arm-m/CMSIS/Include/core_sc300.h
index d66621031..ce1405f9f 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_sc300.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_sc300.h
@@ -1815,7 +1815,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_starmc1.h b/external/arch/arm/arm-m/CMSIS/Include/core_starmc1.h
index d86c8d385..a46f65ae4 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_starmc1.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_starmc1.h
@@ -3460,7 +3460,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
